### ISSUE TITLE
Snowflake: Missing parameter ALLOWED_NETWORK_RULE_LIST, BLOCKED_NETWORK_RULE_LIST in CREATE NETWORK POLICY

### ISSUE THREAD

**Issue by moreaupascal56** [sqlfluff/sqlfluff#6821]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Missing parameter ALLOWED_NETWORK_RULE_LIST, BLOCKED_NETWORK_RULE_LIST in CREATE NETWORK POLICY

```
L:   2 | P:   3 |  PRS | Line 2, Position 3: Found unparsable section:
                       | "ALLOWED_NETWORK_RULE_LIST = ('blabla'..."
WARNING: Parsing errors found and dialect is set to 'snowflake'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```
for file 
```sql
CREATE NETWORK POLICY np
  ALLOWED_NETWORK_RULE_LIST = ('blabla','blabla2','blabla3')
  COMMENT='comment'
  ;
```

### Expected Behaviour

sqlfluff lints file 

### Observed Behaviour

see above

### How to reproduce

see above

### Dialect

snowflake

### Version

3.4.0

### Configuration

no config

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
